### PR TITLE
[YUNIKORN-2211] Replace Allocation uuid with allocationID

### PR DESCRIPTION
**Pull request by manirajv06**
### What is this PR for?
Replace all `uuid` references with `allocationID `

### What type of PR is it?
* [ ] - Improvement

### Todos
* [ ] - Task

### What is the Jira issue?
https://issues.apache.org/jira/browse/YUNIKORN-2211

### How should this be tested?
Using existing unit tests

### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

**Comment by pbacsko** [comment]
Need to update `go.mod`.

File: pkg/common/si_helper_test.go [hunk 1]
```diff
@@ -1,19 +1,19 @@
 /*
- Licensed to the Apache Software Foundation (ASF) under one
```
Comment (chia7712) [inline pkg/common/si_helper_test.go:2]: Those changes are unrelated to this PR. Should we address it in another ticket? I guess there are other files having similar issue.

**Review by chia7712 — COMMENTED** [review]
nice patch and just one trivial comment left.

File: pkg/common/si_helper_test.go (hunk 1 above)
Comment (manirajv06) [reply pkg/common/si_helper_test.go:2]: unintended changes because of IDE formatting (I think so). Please check the reason if possible so that we can fix it next time. For now, let it go through for this fix.

**Comment by chia7712** [comment]
@manirajv06 Please take a look at failed test. overall +1

File: Makefile [hunk 2]
```diff
@@ -523,4 +523,6 @@ arch:
 e2e_test: tools
 	@echo "running e2e tests"
 	cd ./test/e2e && \
-	ginkgo -r $(E2E_TEST) -v -keep-going -- -yk-namespace "yunikorn" -kube-config $(KUBECONFIG)
```
Comment (chia7712) [inline Makefile:526]: I guess those changes are used to run test locally. Please revert it before merging :)

**Review by wilfred-s — APPROVED** [review]
LGTM +1
